udata_recommendations/views.py

```diff
@@ -10,12 +10,20 @@
 
 def has_recommendations(ctx):
     dataset = ctx['dataset']
-    return 'recommendations' in dataset.extras and len(dataset.extras['recommendations'])
+    return dataset and dataset.extras.get('recommendations', [])
 
 
 @template_hook('dataset.display.after-description', when=has_recommendations)
 def dataset_recommendations(ctx):
-    reco_ids = ctx['dataset'].extras['recommendations']
-    recommendations = Dataset.objects.filter(id__in=reco_ids)
-    return theme.render('dataset-recommendations.html',
-                        recommendations=recommendations)
+    reco_ids = []
+    recommendations = ctx['dataset'].extras['recommendations']
+
+    # Get at most n unique recommendations
+    # Recommendations are already sorted by score in desc order
+    limit = current_app.config.get('RECOMMENDATIONS_NB_RECOMMENDATIONS')
+    reco_ids = list({r['id']: 0 for r in recommendations})[:limit]
```

Comment:
what is this `: 0`? 🧐 are you building a dict?

Reply:
Yup, explained here and I added a comment https://github.com/opendatateam/udata-recommendations/pull/142#discussion_r470634422